### PR TITLE
U/jrbogart/abs path

### PR DESCRIPTION
**Pull request by JoanneBogart**
Add routine to return absolute path for a dataset.

### CODE REVIEW

**Comment by stuartmcalpine** [comment]
What about storing the `root_dir` in the schema provenance table? The user would then never have to think about the root dir, just the schema name. Also if we ever had to move the data we would just have to update the root dir in the provenance table. And to reconstruct the abs_path we would just need to pull the root dir out of the table metadata.

**Comment by JoanneBogart** [comment]
Storing root_dir in the database itself doesn't cover everything.
- user can set root_dir to something non-standard when they register (and if they have, they better set it to the same thing when they query)
- if we mirror the data (but not the Postgres database) at another site, the stored root_dir would be wrong for users at the alternate site.
I think the best we can do is keep track of a default root_dir per site, with NERSC currently being the only recognized site. And that information should be stored in the code or, probably better, in a configuration file within the package that the code reads.   Then, when we actually have multiple sites, we should also allow users to specify site, probably with an environment variable. This is the technique used in GCRCatalogs.

File: src/dataregistry/query.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
-import os
-from datetime import datetime
+# from datetime import datetime
```
Comment (stuartmcalpine) [inline src/dataregistry/query.py:1]: can remove if not used

File: src/dataregistry/query.py [hunk 2]
```diff
@@ -361,3 +368,35 @@ def gen_filter(self, property_name, bin_op, value):
...
+        # Fetch absolute path, owner type and owner for the dataset
+        if schema != self._schema:
+            raise DataRegistryNYI("schema != default is not yet supported")
+
+        results = self.find_datasets(property_names=['dataset.owner_type',
+                                                     'dataset.owner',
+                                                     'dataset.relative_path'],
+                                     filters=[('dataset.dataset_id', '==',
+                                               dataset_id)])
+        row = results.fetchone()
```
Comment (stuartmcalpine) [inline src/dataregistry/query.py:400]: should this have a check to see if the dataset was found, will crash otherwise

**Comment by stuartmcalpine** [comment]
> Storing root_dir in the database itself doesn't cover everything.
> 
> * user can set root_dir to something non-standard when they register (and if they have, they better set it to the same thing when they query)
> * if we mirror the data (but not the Postgres database) at another site, the stored root_dir would be wrong for users at the alternate site.
>   I think the best we can do is keep track of a default root_dir per site, with NERSC currently being the only recognized site. And that information should be stored in the code or, probably better, in a configuration file within the package that the code reads.   Then, when we actually have multiple sites, we should also allow users to specify site, probably with an environment variable. This is the technique used in GCRCatalogs.

Ok, I will start a new PR with this suggestion. ^^^

I added a quick CI test to test the `get_dataset_absolute_path` function

Just one comment about crashing scenario if that dataset doesnt exist. Other than that approved to merge

File: src/dataregistry/query.py (hunk 1 above)
Comment (JoanneBogart) [reply src/dataregistry/query.py:1]: Done

File: src/dataregistry/query.py (hunk 2 above)
Comment (JoanneBogart) [reply src/dataregistry/query.py:400]: Yes, thanks for finding this.
Should it raise an exception or just issue a warning? I'll go with the latter for now.

**Comment by JoanneBogart** [comment]
Thanks for adding the CI test.

File: src/dataregistry/query.py (hunk 2 above)
Comment (stuartmcalpine) [reply src/dataregistry/query.py]: I think since it will crash anyway, it might as well be an exception to make it clear the reason for the crash

**Review by stuartmcalpine — APPROVED** [review]

**Pull request merged**